File: python/python-ds-practice/34_same_frequency/same_frequency.py
def freq_counter(col):
    freq = {}
    for elem in col:
        freq[elem] = freq.get(elem, 0) +1
    return freq


def same_frequency(num1, num2):
    """Do these nums have same frequencies of digits?
    
        >>> same_frequency(551122, 221515)
        True
        
        >>> same_frequency(321142, 3212215)
        False
        
        >>> same_frequency(1212, 2211)
        True
    """
    # Convert numbers to strings so we can iterate over digits
    str1, str2 = str(num1), str(num2)
    
    # Use freq_counter to count the frequency of each digit
    freq1, freq2 = freq_counter(str1), freq_counter(str2)
    
    # Check if the frequencies of digits are the same
    return freq1 == freq2

File: prob-freq-count-pointers/same-frequency.js
// add whatever parameters you deem necessary
function sameFrequency(num1, num2) {
  // convert to strings and compare to see if they have the same frequency helping with time complexity
  let str1 = num1.toString();
  let str2 = num2.toString();

  // if strings are not the same length, return false
  if (str1.length !== str2.length) return false;

  // ojects to hold digit frequency
  let count1 = {};
  let count2 = {};

  // for each char in str1, add to count1
  for (let char of str1) {
    count1[char] = (count1[char] || 0) + 1;
  }

  // for each char in str2, add to count2
  for (let char of str2) {
    count2[char] = (count2[char] || 0) + 1;
  }

  // compare count1 and count2 keys and values
  for (let key in count1) {
    if (count1[key] !== count2[key]) return false;
  }
  // return true if all keys and values are the same
  return true;
}
